### PR TITLE
[WIP] Update Raven Simple Sample to v6

### PR DESCRIPTION
**Pull request by kbaley**
This is pending some of the latest changes in NServiceBus.RavenDB so doesn't work yet. Still some discussion on how to access the Raven session from message handlers.

### CODE REVIEW

File: samples/ravendb/simple/Version_6/Server/ShipOrderHandler.cs [hunk 1]
```diff
@@ -6,17 +7,10 @@
...
-    {
-        this.sessionProvider = sessionProvider;
-    }
-
-    public void Handle(ShipOrder message)
+    public async Task Handle(ShipOrder message, IMessageHandlerContext context)
     {
-        var session = sessionProvider.Session;
-        session.Store(new OrderShipped
+        var session = context.GetRavenSession();
```
Comment (kbaley) [inline samples/ravendb/simple/Version_6/Server/ShipOrderHandler.cs]: Still need to work out details about how users will access the Raven session from a handler

File: samples/ravendb/simple/Version_6/Client/app.config [hunk 2]
```diff
@@ -8,4 +8,4 @@
       <add Assembly="Shared" Endpoint="Samples.RavenDB.Server"/>
     </MessageEndpointMappings>
   </UnicastBusConfig>
-</configuration>
+<startup><supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.5.2"/></startup></configuration>
```
Comment (SimonCropp) [inline samples/ravendb/simple/Version_6/Client/app.config]: remove this

File: samples/ravendb/simple/Version_6/Server/app.config [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (SimonCropp) [inline samples/ravendb/simple/Version_6/Server/app.config]: delete this file

**Comment by johnsimons** [comment]
Duplicate of #1121

**Pull request closed without merging**